README.md

```diff
@@ -20,6 +20,10 @@ If you'd like to help out, please see [CONTRIBUTING.md](CONTRIBUTING.md).
 
 ## Release history
 
+* Release 0.21.4 (2023-07-07)
```

Comment:
I think we cherry-pick the release history removal so we can rely on new-style release notes for this?

Reply:
Done.